client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/pages/SelfConfirmPasswordReset.java

```diff
@@ -69,7 +69,7 @@
     public SelfConfirmPasswordReset(final PageParameters parameters) {
         super(parameters);
 
-        if (parameters == null || parameters.get("token").isEmpty()) {
+        if (parameters != null || parameters.get("token").isEmpty()) {
```

Comment:
Why changing this check? Please revert.

Reply:
Original:

```java
        if (parameters == null || parameters.get("token").isEmpty()) {
            LOG.debug("No token parameter found in the request url");
            parameters.add("errorMessage", getString("self.confirm.pwd.reset.error.empty")); // No good. NPE if parameters is `null`
            setResponsePage(getApplication().getHomePage(), parameters);
        }
```

See https://github.com/apache/syncope/blob/master/client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/pages/SelfConfirmPasswordReset.java#L72

Reply:
The original check is like "if parameters is null, or, if not, if there is a token"; your change is like "if parameters is not null and there is a token".

I don't understand why you are  proposing to change the check.

Reply:
If the parameters is null, the condition and the flow continues on where you have a chance of NPE. The change attempts to fix that, but you're right in that I am not quite sure what the behavior should be here. Clearly, if the `parameters` is null, we can't add anything to it, but what should happen if it's null anyway? can it be null at all?

Reply:
Perhaps:

```
if (parameters != null && parameters.get("token").isEmpty()) {
}
```

Better?

Reply:
I was missing the `parameters.add()` inside the if block; at this point your latest proposal is the best one, I agree.

Reply:
Sorry, I was meaning: let's go ahead with your original proposal, let's resolve this.